src/databricks/labs/remorph/reconcile/connectors/databricks.py

```diff
@@ -1,16 +1,33 @@
+from pyspark.errors import PySparkException
 from pyspark.sql import DataFrame
+from pyspark.sql.functions import col
 
 from databricks.labs.remorph.reconcile.connectors.data_source import DataSource
 from databricks.labs.remorph.reconcile.recon_config import Schema, Tables
 
 
 class DatabricksDataSource(DataSource):
     def read_data(self, schema_name: str, catalog_name: str, query: str, table_conf: Tables) -> DataFrame:
-        # Implement Databricks-specific logic here
-        return NotImplemented
+        try:
+            df = self.spark.sql(query)
+            df = df.select([col(column).alias(column.lower()) for column in df.columns])
+            return df
+        except PySparkException as e:
+            error_msg = f"An error occurred while fetching Databricks Data using the following {query} in DatabricksDataSource : {e!s}"
+            raise PySparkException(error_msg) from e
 
     def get_schema(self, table_name: str, schema_name: str, catalog_name: str) -> list[Schema]:
-        # Implement Databricks-specific logic here
-        return NotImplemented
+        try:
+            full_table_name = (
+                f"{catalog_name}.{schema_name}.{table_name}" if catalog_name else f"{schema_name}.{table_name}"
+            )
+            schema_df = self.spark.sql(f"describe table {full_table_name}").where("col_name not like '#%'").distinct()
```

Comment:
we should use system tables, that we will enforce usage on UC

Reply:
Yes, Using information schema for all the sources except databricks.